modules/python/python-options.c

```diff
@@ -73,14 +77,31 @@ python_option_create_value_py_object(const PythonOption *s)
   return value;
 }
 
+PythonOption *
+python_option_ref(PythonOption *s)
+{
+  g_assert(!s || g_atomic_counter_get(&s->ref_cnt) > 0);
+
+  if (s)
+    {
+      g_atomic_counter_inc(&s->ref_cnt);
+    }
+  return s;
+}
```

Comment:
Why do need reference counting for PythonOption.
The ownership of this should be something simple in my opinion.
A PythonOption should be owned by its container class (PythonOptions), which should belong to a driver.

Accumulating multiple options() block can also be done without creating multiple Options instances, as far as I see (`python_options_add_options()` sounds weird)

Reply:
The problem is that a `PythonOptions` does not necessary belong to one driver. When we clone a parser we need to either clone `PythonOptions` with all its `PythonOption`, or make `PythonOptions` and `PythonOption` refcounted, or clone `PythonOptions` and make `PythonOption` refcounted. I did the latter.

Reply:
Cloning `PythonOption` is overkill, as we need to write the clone fn for all types. We can fortunately refcount it as it does not change after constructing it.